executor/src/lib.rs

```diff
@@ -45,122 +38,86 @@ pub type SerializedTransactionDigest = u64;
 
 #[async_trait]
 pub trait ExecutionState {
-    /// The type of the transaction to process.
-    type Transaction: DeserializeOwned + Send + Debug;
-
-    /// The error type to return in case something went wrong during execution.
-    type Error: ExecutionStateError;
-
-    /// The execution outcome to output.
-    type Outcome;
-
     /// Execute the transaction and atomically persist the consensus index. This function
     /// returns an execution outcome that will be output by the executor channel. It may
     /// also return a new committee to reconfigure the system.
     async fn handle_consensus_transaction(
         &self,
         consensus_output: &ConsensusOutput,
         execution_indices: ExecutionIndices,
-        transaction: Self::Transaction,
-    ) -> Result<Self::Outcome, Self::Error>;
-
-    /// Simple guardrail ensuring there is a single instance using the state
-    /// to call `handle_consensus_transaction`. Many instances may read the state,
-    /// or use it for other purposes.
-    fn ask_consensus_write_lock(&self) -> bool;
-
-    /// Tell the state that the caller instance is no longer using calling
-    //// `handle_consensus_transaction`.
-    fn release_consensus_write_lock(&self);
+        transaction: Vec<u8>,
+    );
 
     /// Load the last consensus index from storage.
-    async fn load_execution_indices(&self) -> Result<ExecutionIndices, Self::Error>;
+    async fn load_execution_indices(&self) -> ExecutionIndices;
 }
 
 /// The output of the executor.
-pub type ExecutorOutput<State> = (
-    SubscriberResult<<State as ExecutionState>::Outcome>,
-    SerializedTransaction,
-);
+pub type ExecutorOutput = (SubscriberResult<()>, SerializedTransaction);
 
 /// A client subscribing to the consensus output and executing every transaction.
 pub struct Executor;
 
 impl Executor {
     /// Spawn a new client subscriber.
-    pub async fn spawn<State>(
-        store: Store<(CertificateDigest, BatchDigest), Batch>,
-        execution_state: Arc<State>,
+    pub fn spawn<State>(
+        name: PublicKey,
+        network: oneshot::Receiver<P2pNetwork>,
```

Comment:
Why isn't the network able to be directly passed in and instead is a oneshot?

Reply:
Ah, ok I see. The executor is started outside the primary and the network is constructed inside the primary which makes the dependency difficult to have at the time of construction.